### ISSUE TITLE
iphone errors

### ISSUE THREAD

**Issue by pdxbenjamin** [wojodesign/simplecart-js#253]
simplecart doesn't seem to work on iphone?
I get, with the safari iphone debugger:
Javascript: Error
undefined
QUOTA_EXCEEDED_ERR: DOM Exception 22: An attempt was made to add something to storage.?

**Comment by pdxbenjamin** [comment]
never mind, I had private browsing on.

**Comment by brettwejrowski** [comment]
Yeah I ran into this same issue when testing on browser stack with older iOS versions and was wondering what the issue is. Thanks for the clarity.  We run our test suite against iOS and it should work just fine.

**Comment by nternetinspired** [comment]
Apparently, this is a Safari issue (iOS or OSX) which is by design. When in private browsing mode localstorage appears to be available but a call to .setItem throws this error.

I haven't dug around to try and find where this is set (I am not a coder! If I knew what the hell I was doing I would create a PR of course.) but I believe the fix given here may resolve this issue: https://github.com/jayphelps/history.js/commit/ddbf86ce4e7fccb6ae12555831205b88375c65fc 

It would be nice to have simplecart work in private browsing mode of course ;) Thanks.

**Comment by brettwejrowski** [comment]
I'm not sure how to handle this case, I guess I can try to actually store something and catch the error, where I could use cookies to polyfill... but iOS users can choose to reject cookies as well.  I will have to think about this one.  If anyone has suggestions, let me know

**Comment by nternetinspired** [comment]
Approach 2 tested and confirmed on iOS5.1.1. with private browsing enabled and disabled. Cart now functioning as expected. 

Thanks Roberto.
